code/modules/reagents/chemistry/reagents/medicine_reagents.dm

```diff
@@ -1105,3 +1105,64 @@
 	id = "corazone"
 	description = "A medication used to treat pain, fever, and inflammation, along with heart attacks."
 	color = "#F5F5F5"
+
+/datum/reagent/medicine/modafinil
+	name = "Modafinil"
+	id = "modafinil"
+	description = "Long-lasting sleep suppressant that very slightly reduces stun and knockdown times. Overdosing has horrendous side effects and deals lethal oxygen damage, will knock you unconscious if not dealt with."
+	reagent_state = LIQUID
+	color = "#BEF7D8" // palish blue white
+	metabolization_rate = 0.1 * REAGENTS_METABOLISM
+	overdose_threshold = 10 // low overdose treshold due to very low and random metabolism rate
+	taste_description = "salt" // it actually does taste salty
+	var/overdose_progress = 0 // to track overdose progress
+
+/datum/reagent/medicine/modafinil/on_mob_life(mob/living/M)
+	if(!overdosed) // We do not want any effects on OD
+		M.AdjustStun(-5, 0)
+		M.AdjustKnockdown(-5, 0)
+		M.AdjustUnconscious(-5, 0)
+		M.adjustStaminaLoss(-0.5*REM, 0)
+		M.SetSleeping(0, FALSE)
```

Comment:
if #34664 is merged you could make a SLEEPIMMUNE trait

Reply:
nice, I'll put a reminder down for that 👍

Reply:
Actually, SLEEPIMMUNE seems to be sort of required as M.SetSleeping(0, FALSE) doesn't seem to do it, meaning cafe_latte has basically no effect at all on sleeping toxins, only against single one-time sleeps...